docs/NETWORK-ACLS.md

```diff
@@ -0,0 +1,289 @@
+  Can be combined with other match types except for `eidset` (see [limitations](#limitations)).
+
+* `eidset`: special match type for the overlay network. Matches IPs of all applications deployed in the same network
+  as well as all IPs with statically configured DNS entries (under the config field `NetworkInstanceConfig.Dns`).
+  For this type, `value` field is not used.
+  Can be combined with other match types except for `host` (see [limitations](#limitations)).
+
+* `protocol`: `value` should specify the protocol to match.
+  Protocol can be one of `tcp`, `udp`, `icmp`, or `all`, or it can be a numeric value, representing one of these
+  protocols or a different one. A protocol name from `/etc/protocols` is also allowed.
+  Protocol match can be combined with any other match type (often combined with port numbers).
+
+* `lport`: `value` should be an application *local* port number. For filtering actions, this is the source port for outbound
+  traffic and destination port for inbound traffic. For `PORTMAP` action, this represents application port as exposed
+  to the external network (i.e., if `<edge-node-ip>:2222` is mapped to `<app-ip>:22`, `lport` refers to `2222`).
+  `lport` can be combined with any other match type. It is actually required to combine `lport` and `protocol` inside
+  the same ACE.
+  In other words, port without protocol is not valid.
+
+* `fport`: `value` should be a remote endpoint port number (*foreign* port). Used for filtering actions,
+  but not for `PORTMAP` (do not confuse with `lport`, which is still used to represent the forwarded port - the forwarded
+  port is still considered as local).
+  `fport` can be combined with any other match type. It is actually required to combine `lport` with `protocol` inside
+  the same ACE.
+  In other words, port without protocol is not valid.
+
+## Limitations
+
+Here is a summary of all limitations of the current ACL implementation:
+
+* even though the configuration model allows to assign multiple actions to a single ACE, the current implementation
+  expects only one action and rejects any combination of actions with an error message
+
+* match types `eidset` and `host` are not supported for switch network. This is because these match types work
+  in combination with the local DNS service, which is not used for switch networks.
+
+* `host` match type does not work for domains with statically configured DNS entries. Consider using `eidset` instead,
+  which, however, matches *all* statically configured domains and also IPs of all apps in the same network.
+
+* flows dropped by ACLs within switch networks are not flow-logged. This is because denied flows which are not routed,
+  and only forwarded, get dropped before a conntrack entry is created, which is needed for a flowlog entry
+  to be constructed.
+
+* currently `DROP` action is not implemented for local networks. With the implicit reject-all ACE at the end of every ACL,
```

Comment:
I don't understand. We have a default-DROP don't we?

Reply:
Is this about an explicit DROP action?

Reply:
Yeah, this is the default ACE action (ACE configured without specifying the action explicitly).
Drop-all is applied when no ACE matches the flow.

Reply:
So if the API specifies the action of DROP, what does the code do? Does it ignore the rule? Error? Treat as ACCEPT? (We might want to create a story to fix things if the current behavior is surprising.)

Reply:
It is treated as ACCEPT, so we should probably create a story and fix it.
But note the zedcontrol does not allow to add DROP ACE, so with this controller it cannot be reproduced anyway.

Reply:
Bug story created for this issue.